src/api-sdks/connection-api/clients/python/examples-pip/calculate-hss/calculate-hss.py

```diff
@@ -11,76 +11,63 @@
 print(project_file_path)
 
-try:
-    # Create client attached to already running service
-    attacher = connection_api_service_attacher.ConnectionApiServiceAttacher(configuration)
-    api_client = attacher.create_api_client()
-
-    # Open project
-    with open(project_file_path, 'rb') as file:
-            byte_array = file.read()
-    uploadRes = api_client.project.open_project(idea_con_file=byte_array, _content_type='multipart/form-data')
-    api_client.project_id = uploadRes.project_id
-
-    # Get the project data
-    project_data = api_client.project.get_project_data(api_client.project_id)
-    pprint(project_data)
-
-    # Get list of all connections in the project
-    connections_in_project = api_client.connection.get_connections(api_client.project_id)
-
-    # first connection in the project 
-    connection1 = connections_in_project[0]
-    pprint(connection1)
-
-    # run stress-strain CBFEM analysis for the connection id = 1
-    calcParams = ideastatica_connection_api.ConCalculationParameter() # ConCalculationParameter | List of connections to calculate and a type of CBFEM analysis (optional)
-    calcParams.connection_ids = [connection1.id]
-
-    # run stress-strain analysis for the connection
-    con1_cbfem_results = api_client.calculation.calculate(api_client.project_id, calcParams)
-    pprint(con1_cbfem_results)
-
-    # TODO - fix serialization JSON
-    
-    # get detailed results. Results are list of strings
-    # the number of strings in the list correspond to the number of calculated connections
-    results_text = api_client.calculation.get_raw_json_results(api_client.project_id, calcParams)
-    firstConnectionResult = results_text[0];
-    pprint(firstConnectionResult)
-
-    raw_results = json.loads(firstConnectionResult)
-    pprint(raw_results)
-
-    detailed_results = api_client.calculation.get_results(api_client.project_id, calcParams)
-    pprint(detailed_results)
-
-    # get connection setup
-    connection_setup =  api_client.project.get_setup(api_client.project_id)
-    pprint(connection_setup)
-
-    # modify setup
-    connection_setup.hss_limit_plastic_strain = 0.02
-    modifiedSetup = api_client.project.update_setup(api_client.project_id, connection_setup)
-
-    # recalculate connection
-    recalculate_results = api_client.calculation.calculate(api_client.project_id, calcParams)
-    pprint(recalculate_results)
-
-except Exception as e:
-    print("Operation failed : %s\n" % e)
-finally:
-        # Close project
-    api_client.project.close_project(api_client.project_id)
-    api_client.project = None
+
+# Create client attached to already running service
+with connection_api_service_attacher.ConnectionApiServiceAttacher(baseUrl).create_api_client() as api_client:
+    try:
+        # Open project
+        uploadRes = api_client.project.open_project_from_filepath(project_file_path)
+        api_client.project_id = uploadRes.project_id
```

Comment:
Hi @OndraZitko, no this should be set on the project extension class when we open the file also when we import from iom. https://github.com/idea-statica/ideastatica-public/blob/e2856fa1dcdf8dc63faa8dca8fcd9d676d34ff40/src/api-sdks/connection-api/clients/csharp/src/IdeaStatiCa.ConnectionApi/ApiExt/ProjectApiExt.cs#L80.

Reply:
Add to ProjectExt

```
class ProjectExtApi(ProjectApi):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.active_projec_data: Optional[ConProject] = None
        # Add any additional initialization here

    @property
    def project_id(self):
        if self.active_projec_data is None:
            return None
        return self.active_projec_data.project_id
```